torch/distributed/algorithms/quantization/quantization.py

```diff
@@ -85,6 +83,52 @@ def _dequantize_tensor_list(tensor_list, qtype, quant_loss=None):
     dequantized_tensor_list = [_dequantize_tensor(t, qtype) for t in tensor_list]
     return dequantized_tensor_list
 
+def _quantized_a2a_single_handler(*args, **kwargs):
+    group = kwargs.get('group', None)
+    async_op = kwargs.get('async_op', False)
+    qtype = args[0]
+    tensors = args[1]
+    input_tensor = args[2]
+    out_splits = kwargs.get('out_splits', None)
+    in_splits = kwargs.get('in_splits', None)
+    quant_loss = kwargs.get('quant_loss', None)
+    input_tensors = _quantize_tensor(input_tensor, qtype)
+    out_tensors = _quantize_tensor(tensors, qtype)
+    dist.all_to_all_single(out_tensors, input_tensors, out_splits, in_splits, group=group)
+    for i, t in enumerate(_dequantize_tensor(out_tensors, qtype, quant_loss=quant_loss)):
+        tensors[i] = t
+
+def _quantized_a2a_handler(*args, **kwargs):
+    group = kwargs.get('group', None)
+    async_op = kwargs.get('async_op', False)
+    quant_loss = kwargs.get('quant_loss', None)
+    qtype = args[0]
+    tensors = args[1]
+    input_tensor = args[2]
+    input_tensors = _quantize_tensor_list(input_tensor, qtype)
+    out_tensors = _quantize_tensor_list(tensors, qtype)
+    dist.all_to_all(out_tensors, input_tensors, group=group, async_op=async_op)
+    for i, t in enumerate(_dequantize_tensor_list(out_tensors, qtype, quant_loss=quant_loss)):
+        tensors[i] = t
+
+def _quantized_all_gather_handler(*args, **kwargs):
```

Comment:
I feel like these 3 private functions can be written as a single one:

```
def _quantize_handler(quant_fn, dequant_fn, collective_fn, *args, **kwargs)
```

and you can pass in the `quant_fn`, `dequant_fn`, and `collective_fn` which are basically the 3 things that I can see distinguishing these functions in the implementation.

Once you have that, you can define names using `functools.partial`, for example:
```
_quantized_all_gather_handler = partial(_quantize_handler, _quantize_tensor_list, _dequantize_tensor_list, dist.all_to_all)
```

and so on. Finally you can update your `dispatch` dict to use these above names. The only tricky thing I see would be how to handle the output_splits and input_splits arguments in the a2a single handler.